web/src/app/browser/src/contextManager.ts

```diff
@@ -338,21 +338,40 @@ export default class ContextManager extends ContextManagerBase<BrowserConfigurat
     } catch(err) {
       // non-embedded:  if keyboard activation failed, deactivate the keyboard.
 
-      // Make sure we don't infinite-recursion should the deactivate somehow fail.
-      if(this.config.hostDevice.touchable) {
-        // Fallback behavior - if on a touch device, we need to keep a keyboard visible.
-        const defaultStub = this.keyboardCache.defaultStub;
-        if(defaultStub.id != keyboardId || defaultStub.langId != languageCode) {
-          await this.activateKeyboard(defaultStub.id, defaultStub.langId, true).catch(() => {});
-        } // else "We already failed, so give up."
+      const fallback = async () => {
+        // Make sure we don't infinite-recursion should the deactivate somehow fail.
+        if(this.engineConfig.hostDevice.touchable) {
+          // Fallback behavior - if on a touch device, we need to keep a keyboard visible.
+          const defaultStub = this.keyboardCache.defaultStub;
+          if(defaultStub.id != keyboardId || defaultStub.langId != languageCode) {
+            await this.activateKeyboard(defaultStub.id, defaultStub.langId, true).catch(() => {});
+          } // else "We already failed, so give up."
+        } else {
+          // Fallback behavior - if on a desktop device, the user still has a physical keyboard.
+          // Just clear out the active keyboard & OSK.
+          await this.activateKeyboard('', '', false).catch(() => {});
+        }
+      }
+
+      this.engineConfig.signalUser?.wait(); // clear the wait message box, either way.
+
+      const message = (err as Error)?.message ||
+                      'Sorry, the ' + keyboardId + ' keyboard for ' + languageCode + ' is not currently available.';
+
+      if(err instanceof KeyboardScriptError) {
+        // We get signaled about error log messages if the site is connected to our Sentry error reporting
+        // system; we want to know if we have a broken keyboard that's been published.
+        console.error(err || message);
       } else {
-        // Fallback behavior - if on a desktop device, the user still has a physical keyboard.
-        // Just clear out the active keyboard & OSK.
-        await this.activateKeyboard('', '', false).catch(() => {});
+        // If it's just internet connectivity or "file not found" issues, that's not worth reporting
+        // to Sentry.
+        console.warn(err || message);
```

Comment:
The original source being restored here:

https://github.com/keymanapp/keyman/blob/2a4fe24370e8e7fc046ec2aea950332b8778729d/web/src/engine/main/keyboards/kmwkeyboards.ts#L647

https://github.com/keymanapp/keyman/blob/2a4fe24370e8e7fc046ec2aea950332b8778729d/web/src/engine/main/keyboards/kmwkeyboards.ts#L658-L666

These are triggered from the following:

https://github.com/keymanapp/keyman/blob/2a4fe24370e8e7fc046ec2aea950332b8778729d/web/src/engine/main/keyboards/kmwkeyboards.ts#L674

(The one above uses a default message and effectively defaults to `msgType = 'warn'`.)

https://github.com/keymanapp/keyman/blob/2a4fe24370e8e7fc046ec2aea950332b8778729d/web/src/engine/main/keyboards/kmwkeyboards.ts#L743-L746

https://github.com/keymanapp/keyman/blob/2a4fe24370e8e7fc046ec2aea950332b8778729d/web/src/engine/main/keyboards/kmwkeyboards.ts#L796-L799